Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1313,6 +1365,34 @@ public class ChatChannelController: DataController, DelegateCallable, DataStoreP
     }
 }
 
+public class EphemeralMessageEditor {
+    private unowned let channelController: ChatChannelController
+    private var messageId: MessageId?
+
+    init(channelController: ChatChannelController) {
+        self.channelController = channelController
+    }
+
+    public func updateMessage(text: String) {
+        if let messageId {
+            channelController.updateEphemeralMessage(id: messageId, text: text)
+            return
+        }
+        
+        let group = DispatchGroup()
+        group.enter()
+        channelController.createEphemeralMessage(text: text) {
+            self.messageId = try? $0.get()
+            group.leave()
+        }
+    }
+
+    public func publish() {
+        messageId.map { channelController.publishEphemeralMessage(id: $0) }
+        messageId = nil
+    }
```

Comment:
It could be useful to know when it is published, for that we could the use our new `waitForAPIRequest`.